src/Truncate.js

```diff
@@ -0,0 +1,41 @@
+import styled from 'styled-components'
+import {variant, maxWidth} from 'styled-system'
+import PropTypes from 'prop-types'
+import {TYPOGRAPHY, COMMON} from './constants'
+import theme from './theme'
+
+const truncateVariants = variant({
+  variants: {
+    target: {
```

Comment:
I don't love the name `target` for this variant, as it doesn't feel super descriptive. Really, it's an `inline` modifier for when you're nesting truncated text in another element (probably other text). But `target` was used in the [original spec](https://primer.style/css/components/truncate#truncate-target), so I stuck with it.

Reply:
@emplums What have we done with things like this in the past? I also am not a huge fan of `target` as a name for this; I would much prefer a boolean `inline` prop e.g

```jsx
  Some text with a{' '}
  <Truncate as="strong" inline title="branch-name-that-is-really-long">
    branch-name-that-is-really-long
  </Truncate>
```

It feels like the `target` thing is an implementation detail of the CSS version of this component and I would be in favor of changing it here.

Reply:
After looking at other components, the `inline` boolean prop is a pattern in `CircleBadge` and `ProgressBar`. Apologies for not seeing that earlier. The only difference here would be that `inline` would also add `vertical-align: top`. But that's probably fine. I can make an update to follow that pattern.

Reply:
@BinaryMuse @alanbsmith Whenever we run into something that has a certain naming convention/API in Primer CSS that doesn't feel quite right or ergonomic definitely take the liberty to change it! :) That's part of the reason why we won't ever necessarily have a 1:1 mirror between the different language implementations we've got (CSS, React, Rails) - always better to improve the API than stick with something less than ideal for consistency sake + some languages might have features that make a specific API easier than in other languages so it's fine to deviate a bit :)

Reply:
Sounds great. 😄 Updated to `inline` as a boolean prop.